### DISCUSSION TITLE
failed to fetch https://kubernetes-charts.storage.googleapis.com/index.yaml

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [nebari-dev/nebari#218] [Q&A] [answered]
I tried to destroy an old qhub deployment using:
```
cd infrastructure
terraform destroy -auto-approve
```
and got
```
module.qhub.module.kubernetes-dask-gateway.helm_release.dask-gateway: Refreshing state... [id=dask-gateway]

Warning: Interpolation-only expressions are deprecated

  on .terraform\modules\qhub\modules\kubernetes\services\meta\qhub\main.tf line 16, in module "kubernetes-jupyterhub":
  16:           "${var.general-node-group.key}" = var.general-node-group.value

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 3 more similar warnings elsewhere)


Error: failed to fetch https://kubernetes-charts.storage.googleapis.com/index.yaml : 403 Forbidden

  on .terraform\modules\kubernetes-autoscaling\modules\kubernetes\services\cluster-autoscaler\main.tf line 7, in data "helm_repository" "autoscaler":
   7: data "helm_repository" "autoscaler" {


Releasing state lock. This may take a few moments...
```

Update 12/30/2020:

I just tried adding a user to our qhub at jupyter.qhub.esipfed.org and got the same error:
```
rror: failed to fetch https://kubernetes-charts.storage.googleapis.com/index.yaml : 403 Forbidden

  on .terraform/modules/kubernetes-autoscaling/modules/kubernetes/services/cluster-autoscaler/main.tf line 7, in data "helm_repository" "autoscaler":
   7: data "helm_repository" "autoscaler" {


Releasing state lock. This may take a few moments...rror: failed to fetch https://kubernetes-charts.storage.googleapis.com/index.yaml : 403 Forbidden

  on .terraform/modules/kubernetes-autoscaling/modules/kubernetes/services/cluster-autoscaler/main.tf line 7, in data "helm_repository" "autoscaler":
   7: data "helm_repository" "autoscaler" {

```

**Comment by rsignell-usgs** [comment]
Looks like the answer is here: https://stackoverflow.com/a/65404574/2005869

**Comment by tylerpotts** [comment]
@rsignell-usgs I brought this up to the team. We're going to do a sprint tomorrow to work on this. Issue number will be posted here shortly

**Reply by laisbsc** [reply]
Hey @rsignell-usgs! Hope you enjoyed the holidays!

Here we go: the issue number for this ticket is #225. And the epic for removing Helm all over (just in case you are interested) is #224.

Thanks for reporting.

**Comment by tylerpotts** [comment]
@rsignell-usgs I merged your pull request here: https://github.com/Quansight/qhub-terraform-modules/pull/29

Before merging that request I tested it locally and was unable to reproduce the problem. I'm using Helm v3.2.4 and Terraform v0.13.5. While v2.x of Helm may work, we can't guarantee anything below v3.x. This is mainly because v3.x has a security advantage over v2.x

We're also doing our best to eliminate our dependency on Helm for a host of reasons (dask gateway upgrade, minimize the # of dependencies...) which would eliminate this problem in the future

**Reply by rsignell-usgs** [reply]
@tylerpotts,  I just checked my qhub conda environment and I didn't even have the helm package installed.    
It turned out that old helm version was coming from some old openshift testing I did on windows from nearly 2 years ago. 
Yipes, I had no idea!    

(1) Should I just try installing the latest helm package (3.2.4)?

Regarding terraform, @jkellndorfer and I have in our notes that we needed to pin to Terraform=0.13.4 to avoid errors about conflicts.   But that was from several months ago.  I see that the lastest on conda-forge is now terraform=0.14.3.  

(2) What is the suggested upgrade procedure for terraform?

**Reply by costrouc** [reply]
Yes I would install the latest helm package. There are issues moving from helm 2 -> helm 3 if I remember right regarding the `~/.helm` directory.

> (2) What is the suggested upgrade procedure for terraform?

Terraform does not allow you to downgrade once you have upgraded. But in general there should be no issues with upgrading.

**Comment by dharhas** [comment]
So should this have been a github issue rather than a discussion?

**Reply by laisbsc** [reply]
> So should this have been a GitHub issue rather than a discussion?

Hmm, good question!

The way I see it is issues are for Bug reports, problems & recommendations on the docs, and for feature requests. All other questions (hence the name `Discussion Q&A`) should go here. 

I am the newbie, though so please feel free to tell me when I get it wrong. 😄

**Comment by laisbsc** [comment, marked as answer]
The answer to this issue is [here](https://github.com/Quansight/qhub/issues/225#issuecomment-756134292). 🎉
